src/platform/unix/mod.rs

```diff
@@ -593,13 +598,15 @@ impl OsIpcOneShotServer {
         unsafe {
             let fd = libc::socket(libc::AF_UNIX, SOCK_SEQPACKET, 0);
             let mut path: Vec<u8>;
+            let dir = TempDir::new().unwrap();
+            let dir_path = dir.path().join("rust-ipc-socket");
+            
             loop {
```

Comment:
With the new approach, you can drop the loop and the `EINVAL` handling I believe: AIUI we only needed to retry, because using `mktemp()` is not atomic. (Someone else might have taken the name between our `mktemp()` and `bind()` calls.) `tempfile` doesn't have that issue.

Might be prudent though to do this change in a separate commit, just in case... (While all your other changes of course will have to be squashed into one before this can be merged.)